lib/rules/no-manual-cleanup.ts

```diff
@@ -122,24 +99,29 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
             });
           }
         } else {
-          defaultRequireFromTestingLibrary = declaratorNode.id;
-        }
-      },
-      'Program:exit'() {
-        if (defaultImportFromTestingLibrary) {
-          const references = context.getDeclaredVariables(
-            defaultImportFromTestingLibrary
-          )[0].references;
-
+          // case: const utils = require('testing-library-module')
+          const references = getVariableReferences(context, declaratorNode);
           reportImportReferences(references);
         }
+      }
+    }
 
-        if (defaultRequireFromTestingLibrary) {
-          const references = context
-            .getDeclaredVariables(defaultRequireFromTestingLibrary.parent)[0]
-            .references.slice(1);
+    return {
+      'Program:exit'() {
```

Comment:
This rule is so much simpler now: it just needs to focus on checks around the node imports, but doesn't need to handle anything to find them.